### DISCUSSION TITLE
Can middleware.ts be moved to another folder when using app router with i18n routing?

### DISCUSSION THREAD

**Discussion by jayporta** [amannn/next-intl#1101] [Q&A]
[The docs](https://next-intl-docs.vercel.app/docs/getting-started/app-router/with-i18n-routing#move-i18n-ts) show how to move the i18n.ts file, but can the middleware.ts file be moved as well? I'd like to keep them together in the same directory for organizing all next-intl files.

**Comment by meJohnnyRobert** [comment]
I had mine at the root of my project before moving it inside the src folder together with the `i18n.ts `and `navigation.ts` and everything seems work just fine, so I think you can move it wherever you like as long as you update the import paths correctly in your other files (VSCode should ask if you want to update the imports for it automatically after you move it)

You can give it a try and check if everything still works, I think it should work just fine.
